streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsProducer.java

```diff
@@ -371,6 +371,7 @@ void flush() {
 
     void close() {
         producer.close();
+        transactionInFlight = false;
```

Comment:
```suggestion
        transactionInFlight = false;
        transactionInitialized = false;
```

Reply:
With this can remove the `transactionInitialized = false`  and replace the `producer.close()` with just  `close()` over in `StreamsProducer#resetProducer`